File: src/examples/changeSignatory/drawable.cs
using Criipto.Signatures;
using Criipto.Signatures.Models;

using (var client = new CriiptoSignaturesClient("{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}")) {
    // Setup document input
    var documents = new List<DocumentInput>{
        new DocumentInput {
            pdf = new PadesDocumentInput {
                title = "Dotnet Sample",
                blob = new Byte[64], // Should be the bytes of a PDF file
                storageMode = DocumentStorageMode.Temporary
            }
        }
    };

    // Setup signature order input
    var createSignatureOrderInput = new CreateSignatureOrderInput
    {
        title = "Dotnet Sample",
        documents = documents,
        evidenceProviders = new List<EvidenceProviderInput>() {
            new EvidenceProviderInput() {
                enabledByDefault = false,
                drawable = new DrawableEvidenceProviderInput() {
                    requireName = true
                }
            }
        }
    };

    // Create signature order
    var signatureOrder = await client.CreateSignatureOrder(createSignatureOrderInput);

    // Add signatory
    var signatory = await client.AddSignatory(
        signatureOrder.id
    );

    // Find drawable evidence provider
    var drawable =
        signatureOrder!.evidenceProviders
            .Where(e => e is DrawableSignatureEvidenceProvider)
            .First();

    // Change signatory to enable drawable
    await client.ChangeSignatory(
        signatory.id,
        new ChangeSignatoryInput() {
            evidenceProviders = new List<SignatoryEvidenceProviderInput>() {
                new SignatoryEvidenceProviderInput() {
                    id = drawable.id
                }
            }
        }
    );
}

File: src/examples/changeSignatory/drawable.py
from criipto_signatures import CriiptoSignaturesSDKSync
from criipto_signatures.models import (
  CreateSignatureOrderInput,
  DocumentInput,
  PadesDocumentInput,
  DocumentStorageMode,
  AddSignatoryInput,
  EvidenceProviderInput,
  DrawableEvidenceProviderInput,
  ChangeSignatoryInput,
  SignatoryEvidenceProviderInput,
)

client = CriiptoSignaturesSDKSync(
  "{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"
)

# Create signature order
signatureOrder = client.createSignatureOrder(
  CreateSignatureOrderInput(
    title="Python sample",
    documents=[
      DocumentInput(
        pdf=PadesDocumentInput(
          title="My document",
          blob=bytes("...", "utf-8"),  # bytes object, or a base64 encoded string
          storageMode=DocumentStorageMode.Temporary,
        )
      )
    ],
    evidenceProviders=[
      EvidenceProviderInput(
        enabledByDefault=False,
        drawable=DrawableEvidenceProviderInput(requireName=True),
      )
    ],
  )
)

# Add signatory
signatory = client.addSignatory(AddSignatoryInput(signatureOrderId=signatureOrder.id))

# Find drawable evidence provider
drawable = signatureOrder.evidenceProviders[0]

# Change signatory to enable drawable
client.changeSignatory(
  ChangeSignatoryInput(
    signatoryId=signatureOrder.id,
    evidenceProviders=[SignatoryEvidenceProviderInput(id=drawable.id)],
  )
)
